rj_gameplay/stp/role/assignment/naive.py

```diff
@@ -132,16 +161,20 @@ def assign_prioritized_roles(
             request_idx: int = request_ind[assignment_idx]
             robot_idx: int = robot_ind[assignment_idx]
             cost: float = robot_costs[robot_idx, request_idx]
-
             # If cost is equal to INVALID_COST, then it means that role assignment
             # failed. Throw an exception.
             if cost == INVALID_COST:
                 raise RuntimeError("Hard constraints not satisfied!")
 
-            robot: stp.rc.Robot = free_robots[robot_idx]
-            role_id = keys_list[request_idx]
+            # If the row index is not the unassigned row, fill the request
+            if robot_idx < unassigned_idx:
+                robot: stp.rc.Robot = free_robots[robot_idx]
+                role_id = keys_list[request_idx]
+                flat_results[role_id].assign(robot, cost)
 
-            flat_results[role_id].assign(robot, cost)
+        # If unassigned index chosen, remove it from the list of row indices, which will be removed from free robots
+        if unassigned_idx in robot_ind:
+            robot_ind = robot_ind[robot_ind != unassigned_idx]
 
         # Remove the robots from free_robots.
```

Comment:
From @kasohrab 's comment above, might be worth changing or expanding on this comment for clarity